search/saddleback_search.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Implementation of saddleback algorithm for 2D arrays.
```

Comment:
Provide a Wikipedia link in Markdown format (if possible/available).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Reply:
Okay :thumbsup: